msgraphcore/middleware/utils.py

```diff
@@ -0,0 +1,31 @@
+import datetime
+from email.utils import parsedate_to_datetime
+
+
+def get_retry_after(response):
+    """
+    Check if retry-after is specified in the response header and get the value
+    """
+    request = response.request
+    retry_after = request.headers.get("retry-after")
+    if retry_after:
+        return parse_retry_after(retry_after)
+    for ms_header in ["retry-after-ms", "x-ms-retry-after-ms"]:
```

Comment:
I've never seen these headers.  Do we need to add these to the retry handler spec for the other clients?

Reply:
No need. These headers are only on the azure side. But since retry handler will come after the authentication handler, they will be handled by the underlying azure identity package.